adapters/repos/db/lsmkv/bucket.go

```diff
@@ -344,6 +344,9 @@ func (b *Bucket) IterateObjects(ctx context.Context, f func(object *storobj.Obje
 }
 
 func (b *Bucket) ApplyToObjectDigests(ctx context.Context, f func(object *storobj.Object) error) error {
+	b.disk.flushVsCompactLock.Lock()
+	defer b.disk.flushVsCompactLock.Unlock()
```

Comment:
could you elaborate (and add comment) why this lock is needed here?

Reply:
Is not needed anymore will remove